### ISSUE TITLE
How to mapping ground truth from color to labelIds

### ISSUE THREAD

**Issue by LemonAniLabs** [TobyPDE/FRRN#14]
Hi @TobyPDE ,

I have some training data was labeled in color, I would like to transform color image to labelIds image for training.
[label.py](https://github.com/TobyPDE/FRRN/blob/master/labels.py#L66-L103) have the mapping table, so I just using 'id' to replace [r,g,b] and save one channel to image.
Is this make sense?

Any idea?
Thanks.

**Comment by TobyPDE** [comment]
There are two possibilities: 

1. You perform the mapping offline and store the result in a numpy array. You can then use the generic data provider that receives two numpy arrays for images and targets.
2. You write your own dataprovider that performs the mapping. 

The code in label.py is only used for visualization.

**Comment by LemonAniLabs** [comment]
I mean using the 'label.id' is correct or not?
I see in [dltools/utility.py](https://github.com/TobyPDE/FRRN/blob/master/dltools/utility.py#L273-L296) is mapping from label.trainId to color instead of label.Id
So if I want to mapping color to label value, should map to trainId?

**Comment by TobyPDE** [comment]
The new code allows mapping from colors to ids if you use the C++ version of Chianti.
